### PR TITLE
Update reagent & react. Build with shadow-cljs

### PR DESCRIPTION
**Pull request by madis**
As part of updating district0x libraries - React API had changed (render
moved from react core to react dom) and also our toolchain - building
tests with shadow-cljs improves the development workflow significantly.

No API changes of that library.

Also added CircleCI config - after merging to master (and in the
presence of ENV variables) will get built & published to Clojars
automatically

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  test:
+    working_directory: ~/ci
+    docker:
+      # Primary container image where all steps run.
+      - image: 487920318758.dkr.ecr.us-west-2.amazonaws.com/cljs-web3-ci:latest
+        aws_auth:
+            aws_access_key_id: $AWS_ACCESS_KEY_ID
+            aws_secret_access_key: $AWS_SECRET_ACCESS_KEY
+      # Secondary container image on common network.
+      - image: trufflesuite/ganache-cli:latest
```
Comment (ginesdt) [inline .circleci/config.yml]: if I'm not mistaken, this project does not use ganache/truffle, then why are you deploying the trufflesuite/ganache-cli:latest image for test?

File: build.cljs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+(ns build
+  (:require [clojure.tools.build.api :as b]
+            [deps-deploy.deps-deploy :as dd]))
+
+(def lib 'district0x/district-ui-reagent-render) ; ends up as <group-id>/<artifact-id> in pom.xml
```
Comment (ginesdt) [inline build.cljs:5]: You are using the group-id district0x. Is that intentional? shouldn't be using io.github.district0x?

File: src/deps.cljs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{:npm-deps
+ {"karma" "6.3.0"
+  "react" "17.0.2"
```
Comment (ginesdt) [inline src/deps.cljs:3]: you're using react v17, but v18 is already out, since you are updating react, doesn't it make sense to use latest version?

File: deps.edn [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        cljs-react-test/cljs-react-test {:mvn/version "0.1.4-SNAPSHOT"}
+        re-frame/re-frame {:mvn/version "1.2.0"} }
+
+ :install-deps true
+ :aliases {:build {:deps {io.github.clojure/tools.build {:tag "v0.8.1" :sha "7d40500"}
+                          slipset/deps-deploy {:mvn/version "RELEASE"}}
+                   :ns-default build}
+           :deploy {:replace-deps {slipset/deps-deploy {:mvn/version "RELEASE"}}
+                    :exec-fn deps-deploy.deps-deploy/deploy
+                    :exec-args {:installer :remote :artifact "target/district-ui-web3-1.3.3.jar"}}}}
```
Comment (ginesdt) [inline deps.edn]: AFAIK, this project is not related to district-ui-web3, is that intentional?

**Review by ginesdt — COMMENTED** [review]
I have some minor comments. Other than that it looks good.

File: .circleci/config.yml (hunk 1 above)
Comment (madis) [reply .circleci/config.yml]: Right, updated
